core/component.js

```diff
@@ -0,0 +1,16 @@
+const default_options = {
+  entity: null
+};
+
+export class Component {
+  constructor(options) {
+    Object.assign(this,  default_options, options);
+    this.name = this.constructor;
```

Comment:
Should this read `this.constructor.name`?

Reply:
Nope. We want to register components with constructors as map keys, not constructor names, so instead of 
```js
const transform = entity.get_component('Transform');
```

we could do

```js
const transform = entity.get_component(Transform); 
//or .get_component(Cervus.components.Transform) for bundled version
```

Reply:
Yes, but on the component instance itself, what do you need `name` for?